This is libpng 893b8113f04d408cc6177c6de19c9889a48faa24, backported to the
version in wxWidgets-3.0

https://github.com/pnggroup/libpng/commit/893b8113f04d408cc6177c6de19c9889a48faa24

--- src/png/pngpriv.h
+++ src/png/pngpriv.h
@@ -342,18 +342,8 @@
     */
 #  include <float.h>
 
-#  if ( ( (defined(__MWERKS__) && defined(macintosh)) || defined(applec) || \
-    defined(THINK_C) || defined(TARGET_OS_MAC) ) && !wxOSX_USE_IPHONE ) || defined(__SC__)
-     /* We need to check that <math.h> hasn't already been included earlier
-      * as it seems it doesn't agree with <fp.h>, yet we should really use
-      * <fp.h> if possible.
-      */
-#    if !defined(__MATH_H__) && !defined(__MATH_H) && !defined(__cmath__)
-#      include <fp.h>
-#    endif
-#  else
-#    include <math.h>
-#  endif
+#  include <math.h>
+
 #  if defined(_AMIGA) && defined(__SASC) && defined(_M68881)
      /* Amiga SAS/C: We must include builtin FPU functions when compiling using
       * MATH=68881
